### PR TITLE
bug/PDPDEVTOOL-3528-VSCODE-The-dependencies-installation-Fails-if-Use…

### PR DESCRIPTION
**Pull request by karl-anthony-ng**
…r-has-JAVA-8

### CODE REVIEW

**Review by vic-nik — APPROVED** [review]

File: packages/vscode-extension/src/service/TranslationKeys.ts [hunk 1]
```diff
@@ -90,7 +90,9 @@ export const EXTENSION_INSTALLATION = {
 export const ERRORS = {
 	NO_ACTIVE_FILE_OR_WORKSPACE: 'ERROR_NO_ACTIVE_FILE_OR_WORKSPACE',
 	NO_ACTIVE_FILE: 'ERROR_NO_ACTIVE_FILE',
-	NO_ACTIVE_WORKSPACE: 'ERROR_NO_ACTIVE_WORKSPACE'
+	NO_ACTIVE_WORKSPACE: 'ERROR_NO_ACTIVE_WORKSPACE',
```
[the remark this answers is not in the document]
Comment (karl-anthony-ng) [reply packages/vscode-extension/src/service/TranslationKeys.ts]: Updated.

File: packages/vscode-extension/src/core/sdksetup/SdkServices.ts [hunk 2]
```diff
@@ -24,9 +24,33 @@ const translationService = new VSTranslationService();
...
-		await install();
-	}
+    await validateJavaVersion();
+
+    if (!fs.existsSync(path.join(sdkPath)) || !(await validateSdk(path.join(sdkPath)))) {
+        await install();
+    }
+}
+
+async function validateJavaVersion() {
```
[the remark this answers is not in the document]
Comment (karl-anthony-ng) [reply packages/vscode-extension/src/core/sdksetup/SdkServices.ts]: Removed await and async for validateJavaVersion().

File: packages/vscode-extension/src/core/sdksetup/SdkServices.ts [hunk 3]
```diff
@@ -24,9 +24,33 @@ const translationService = new VSTranslationService();
...
+    const requiredJavaVersion = `${ApplicationConstants.SDK_REQUIRED_JAVA_VERSION}`;
+    if (installedJavaVersion.startsWith(requiredJavaVersion)) {
+        return;
+    }
+
+    let messageKey;
+    let errorMessage;
+    if (installedJavaVersion === '') {
+        messageKey = `${ERRORS.SDK_JAVA_VERSION_NOT_INSTALLED}`;
+        errorMessage = translationService.getMessage(messageKey, requiredJavaVersion);
```
Comment (davidecorreu) [inline packages/vscode-extension/src/core/sdksetup/SdkServices.ts:46]: We can remove the messageKey variable and directly use `errorMessage = translationService.getMessage(ERRORS.SDK_JAVA_VERSION_NOT_INSTALLED, requiredJavaVersion);`  here and `    errorMessage = translationService.getMessage(ERRORS.SDK_JAVA_VERSION_NOT_COMPATIBLE, installedJavaVersion, requiredJavaVersion);` below in line 51.

Comment (karl-anthony-ng) [reply packages/vscode-extension/src/core/sdksetup/SdkServices.ts]: Updated.

**Review by davidecorreu — APPROVED** [review]

**Pull request merged**